internal/controller/components/kueue/kueue_controller_actions.go

```diff
@@ -33,14 +36,42 @@ func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest)
 	}
 
 	if rConfig || rCluster {
-		return odherrors.NewStopError(MultiKueueCRDMessage)
+		return odherrors.NewStopError(status.MultiKueueCRDMessage)
+	}
+
+	switch kueueCRInstance.Spec.ManagementState {
+	case operatorv1.Managed:
+		if found, err := cluster.OperatorExists(ctx, rr.Client, kueueOperator); err != nil || found {
+			if err != nil {
+				return odherrors.NewStopErrorW(err)
+			}
+
+			return odherrors.NewStopErrorW(ErrKueueOperatorAlreadyInstalled)
+		}
+	case operatorv1.Unmanaged:
+		if found, err := cluster.OperatorExists(ctx, rr.Client, kueueOperator); err != nil || !found {
+			if err != nil {
+				return odherrors.NewStopErrorW(err)
+			}
+
+			return odherrors.NewStopErrorW(ErrKueueOperatorNotInstalled)
+		}
+	default:
+		return nil
```

Comment:
we will still by default to create DSC.components.Kueue.ManagementState to Managed, right? 
so, we are saying, if you use default DSC ^, you cannot have Kueue Operator installed. 
which means, for Managed Cluster , SRE (they only use default DSC) cannot install Kueue Operator.

Reply:
We need to figure out the migration plan for managed cluster